mujoco_warp/_src/ray_test.py

```diff
@@ -0,0 +1,286 @@
+    wp.synchronize()
+    geomid_np = geomid.numpy()[0][0]
+    dist_np = dist.numpy()[0][0]
+    _assert_eq(geomid_np, 5, "geom_id")
+
+    pnt_np, vec_np = pnt.numpy()[0], vec.numpy()[0]
+    unused = np.zeros(1, dtype=np.int32)
+    mj_dist = mujoco.mj_ray(m, d, pnt_np, vec_np, None, 1, -1, unused)
+    _assert_eq(dist_np, mj_dist, "dist-dodecahedron")
+
+  def test_ray_geomgroup(self):
+    """Tests ray geomgroup filter."""
+    m, d, _, _ = test_util.fixture("ray.xml")
+    mujoco.mj_forward(m, d)
+    mx, dx = mjwarp.put_model(m), mjwarp.put_data(m, d)
+
+    # hits plane with geom_group[0] = 1
+    pnt = wp.array([wp.vec3(2.0, 1.0, 3.0)], dtype=wp.vec3)
+    vec = wp.array([wp.normalize(wp.vec3(0.1, 0.2, -1.0))], dtype=wp.vec3)
+    geomgroup = vec6(1, 0, 0, 0, 0, 0)
+    dist, geomid = mjwarp.ray_geom(mx, dx, pnt, vec, geomgroup=geomgroup)
+    wp.synchronize()
+    geomid_np = geomid.numpy()[0][0]
+    dist_np = dist.numpy()[0][0]
+    _assert_eq(geomid_np, 0, "geom_id")
+
+    pnt_np, vec_np = pnt.numpy()[0], vec.numpy()[0]
+    unused = np.zeros(1, dtype=np.int32)
+    mj_dist = mujoco.mj_ray(m, d, pnt_np, vec_np, None, 1, -1, unused)
+    _assert_eq(dist_np, mj_dist, "dist")
+
+    # nothing hit with geom_group[0] = 0
+    pnt = wp.array([wp.vec3(2.0, 1.0, 3.0)], dtype=wp.vec3)
+    vec = wp.array([wp.normalize(wp.vec3(0.1, 0.2, -1.0))], dtype=wp.vec3)
+    geomgroup = vec6(0, 0, 0, 0, 0, 0)
+    dist, geomid = mjwarp.ray_geom(mx, dx, pnt, vec, geomgroup=geomgroup)
+    wp.synchronize()
+    geomid_np = geomid.numpy()[0][0]
+    dist_np = dist.numpy()[0][0]
+    _assert_eq(geomid_np, -1, "geom_id")
+    _assert_eq(dist_np, -1, "dist")
+
+  def test_ray_flg_static(self):
+    """Tests ray flg_static filter."""
+    m, d, _, _ = test_util.fixture("ray.xml")
+    mujoco.mj_forward(m, d)
+    mx, dx = mjwarp.put_model(m), mjwarp.put_data(m, d)
+
+    # nothing hit with flg_static = False
+    pnt = wp.array([wp.vec3(2.0, 1.0, 3.0)], dtype=wp.vec3)
+    vec = wp.array([wp.normalize(wp.vec3(0.1, 0.2, -1.0))], dtype=wp.vec3)
+    dist, geomid = mjwarp.ray_geom(mx, dx, pnt, vec, flg_static=False)
+    wp.synchronize()
+    geomid_np = geomid.numpy()[0][0]
+    dist_np = dist.numpy()[0][0]
+    _assert_eq(geomid_np, -1, "geom_id")
+    _assert_eq(dist_np, -1, "dist")
+
+  def test_ray_bodyexclude(self):
+    """Tests ray bodyexclude filter."""
+    m, d, _, _ = test_util.fixture("ray.xml")
+    mujoco.mj_forward(m, d)
+    mx, dx = mjwarp.put_model(m), mjwarp.put_data(m, d)
+
+    # nothing hit with bodyexclude = 0 (world body)
+    pnt = wp.array([wp.vec3(2.0, 1.0, 3.0)], dtype=wp.vec3)
+    vec = wp.array([wp.normalize(wp.vec3(0.1, 0.2, -1.0))], dtype=wp.vec3)
+    dist, geomid = mjwarp.ray_geom(mx, dx, pnt, vec, bodyexclude=0)
+    wp.synchronize()
+    geomid_np = geomid.numpy()[0][0]
+    dist_np = dist.numpy()[0][0]
+    _assert_eq(geomid_np, -1, "geom_id")
+    _assert_eq(dist_np, -1, "dist")
+
```

Comment:
please add a test for transparent geoms. mjx reference: https://github.com/google-deepmind/mujoco/blob/8e7457d7c6c1cbda42ec39d2297d4f90b3396040/mjx/mujoco/mjx/_src/ray_test.py#L238